### ISSUE TITLE
RenderFlex overflowed using menu with certain shortcuts

### ISSUE THREAD

**Issue by drrnbrns** [flutter/flutter#153105]
### Steps to reproduce

This is a modification of the first sample at https://api.flutter.dev/flutter/material/MenuBar-class.html

1 Run `flutter create --sample=material.MenuBar.1 menubar`
2 In main.dart, replace `_getMenus()` with the code below
3 `flutter run -d chrome`
4 Open the menu to see the RenderFlex warning in the debug output

The key changes are that there's only one menu entry, and it has a shortcut with `LogicalKeyboardKey.arrowRight` (other shortcut keys don't cause the issue).

```

  List<MenuEntry> _getMenus() {
    final List<MenuEntry> result = <MenuEntry>[
      MenuEntry(
        label: 'Menu Demo',
        menuChildren: <MenuEntry>[
          MenuEntry(
            label: 'About',
            shortcut: const SingleActivator(LogicalKeyboardKey.arrowRight),
            onPressed: () {
              showAboutDialog(
                context: context,
                applicationName: 'MenuBar Sample',
                applicationVersion: '1.0.0',
              );
              setState(() {
                _lastSelection = 'About';
              });
            },
          ),
         ],
      ),
    ];
    // (Re-)register the shortcuts with the ShortcutRegistry so that they are
    // available to the entire application, and update them if they've changed.
    _shortcutsEntry?.dispose();
    _shortcutsEntry =
        ShortcutRegistry.of(context).addAll(MenuEntry.shortcuts(result));
    return result;
  }
}
```

### Expected results

No output or warnings expected.

### Actual results

"A RenderFlex overflowed by 7.8 pixels on the right." (see logs for details)

### Code sample

See "Steps to reproduce"

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
(Debugging using Chrome device)

Launching lib\main.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:53109/TC06Nkq1CMg=/ws
Debug service listening on ws://127.0.0.1:53109/TC06Nkq1CMg=/ws
Connecting to VM Service at ws://127.0.0.1:53109/TC06Nkq1CMg=/ws
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during layout:
A RenderFlex overflowed by 7.8 pixels on the right.

The relevant error-causing widget was:
  MenuItemButton MenuItemButton:file:///C:/Projects/menubar/lib/main.dart:31:14

To inspect this widget in Flutter DevTools, visit:
http://127.0.0.1:9101/#/inspector?uri=http%3A%2F%2F127.0.0.1%3A53109%2FTC06Nkq1CMg%3D&inspectorRef=inspector-0

The overflowing RenderFlex has an orientation of Axis.horizontal.
The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
RenderFlex to fit within the available space instead of being sized to their natural size.
This is considered an error condition because it indicates that there is content that cannot be
seen. If the content is legitimately bigger than the available space, consider clipping it with a
ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
like a ListView.
The specific RenderFlex in question is: RenderFlex#41789 relayoutBoundary=up40 OVERFLOWING:
  creator: Row ← _MenuItemLabel ← Align ← Padding ← IconTheme ← Builder ← Listener ←
    RawGestureDetector ← GestureDetector ← Semantics ← DefaultSelectionStyle ← Builder ← ⋯
  parentData: offset=Offset(0.0, 10.0) (can use size)
  constraints: BoxConstraints(0.0<=w<=52.1, 0.0<=h<=Infinity)
  size: Size(52.1, 20.0)
  direction: horizontal
  mainAxisAlignment: spaceBetween
  mainAxisSize: max
  crossAxisAlignment: center
  textDirection: ltr
  verticalDirection: down
◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
════════════════════════════════════════════════════════════════════════════════════════════════════

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
C:\Projects\menubar>flutter doctor -v 
[√] Flutter (Channel stable, 3.22.3, on Microsoft Windows [Version 10.0.19045.4651], locale en-US)
    • Flutter version 3.22.3 on channel stable at C:\Users\User\flutter     
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (3 weeks ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\User\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0--11852314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0--11852314)

[√] VS Code (version 1.92.0)
    • VS Code at C:\Users\User\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.94.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.4651]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 127.0.6533.99
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 127.0.2651.86

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @drrnbrns, it would appear that this has been fixed on the latest stable version (3.24.0). Can you try with this version as well to see if you still experience this issue?

**Comment by drrnbrns** [comment]
After a lot of experimentation, I had to run `flutter downgrade 3.22.3`, as there seem to be a lot of issues around web in 3.24.0.  With 3.24.0, both the original steps and the simpler `flutter create test` steps in my previous comment result in an unstable environment as detailed in my previous comment.

However, there were times when I got 3.24.0 to run and the RenderFlex warning did not appear, so I think we can consider it fixed in that release.  Thanks!
